Use autoload and remove mappings

ddrscott opened the pull request:
- The original mappings overwrote a lot of mines, so I moved them to the README for others to see who the could use the functions/commands.
- created `islime2#iTermSendOperator` which is much more flexible than the original visual mapping
- moved from `plugin` to `autoload` as a best practice.

I'm not expecting this to be merged without a few more tweaks, but figured you might be interested in the operator pending option.

matschaffer commented:
I'm happy to see someone else is getting use out of it :)

It looks pretty different so I'll have to give it a try first but so long as I can still map up my usual keys I should be in good shape.

Thanks!

ddrscott commented:
It doesn't diff well since I split out the mappings into the README, moved the file, and `%s/<SID>/islime2#/g`. A side by side diff maybe more interesting, but in the end. As long as it works, it should be okay.

It's been working good for me.

matschaffer commented:
@ddrscott since your vim chops are obviously better than mine, what's your recommended way to implement these?

```vim
inoremap <leader>cc <Esc>vip"ry:call <SID>iTermSendNext(@r)<CR>
vnoremap <leader>cc "ry:call <SID>iTermSendNext(@r)<CR>
nnoremap <leader>cc vip"ry:call <SID>iTermSendNext(@r)<CR>
```

The idea here is to capture the spirit of vim-slime's `C-c, C-c`

As a subnote, revisiting that plugin for the first time in years and seeing https://github.com/jpalardy/vim-slime#tmux makes me wonder if this whole plugin should maybe be an iTerm2 vim-slime target instead. ;)

ddrscott commented:
I believe I've captured the spirit of the `<Leader>cc` mappings with:

```vim
inoremap <leader>cc <Esc>vip:<C-u>call islime2#iTermSendOperator(visualmode(), 1)<CR>
vnoremap <leader>cc :<C-u>call islime2#iTermSendOperator(visualmode(), 1)<CR>
nnoremap <leader>cc vip:<C-u>call islime2#iTermSendOperator(visualmode(), 1)<CR>
```
The key from my perspective is leave the `"r` register alone. That's the way I would do it.

An alternative is to chain the mappings off of the `islime2#iTermSendOperator` function:

```
nnoremap <silent> <Leader>i :set opfunc=islime2#iTermSendOperator<CR>g@
nmap <leader>cc <leader>iip
vmap <leader>cc <leader>i
imap <leader>cc <Esc><leader>cc
```

Hope that all makes sense. There's too many ways to do things in Vim. The main point of the these alternate mappings is to leave the `"r` register alone and leverage operator pending. Can you try `:help g@` for more info about that.

ddrscott commented:
As for `tmux`, I can use it because I like change my font sizes independently.
At any given point, I have many splits with different font sizes in each split and I like smooth scrolling in each split.

My current setup looks like this:
![image](https://cloud.githubusercontent.com/assets/1975119/26164193/84439f7e-3af1-11e7-847d-5fc02d400839.png)

matschaffer commented:
@ddrscott Thanks! This works great.

Neat workflow!
